File: java/recursions_backtrackings/Day1/BasicRec.java
package recursions_backtrackings.Day1;
import java.util.*;


public class BasicRec {

    /* 
    // (1) Print Name N Times...
    public static void nTimes(int i, int n) {
        if (i > n) return;

        System.out.println("Aman");

        nTimes(i + 1, n);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n = ");
        int n = sc.nextInt();

        nTimes(1, n);

        sc.close();
    }
    */


/* 
    (2) Print Linearly from 1 to n...
    public static void Linearly(int i, int n) {
        if (i > n) return;

        System.out.println(i);
        Linearly(i + 1, n);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n = ");
        int n = sc.nextInt();

        Linearly(1, n);

        sc.close();
    }
*/



/*
    // (3) Print from n to 1...
    public static void ntoOne(int i, int n) {
        if (i < 1) return;

        System.out.println(i);

        ntoOne(i - 1, n);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n = ");
        int n = sc.nextInt();

        ntoOne(n, n);

        sc.close();
    }
*/




/* 
    // (4) Print Lineearly but using backtrack...
    public static void LinearlyBack(int i, int n) {
        if (i < 1) return;

        LinearlyBack(i - 1, n);

        System.out.println(i);
    }


    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n = ");
        int n = sc.nextInt();

        LinearlyBack(n, n);

        sc.close();
    }
*/


    // (5) print from n to 1 using backtrack...
    public static void ntoOneBack(int n) {
        if (n < 1) return;
        
        ntoOneBack(n - 1);
        System.out.println(n);

    }


    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n = ");
        int n = sc.nextInt();

        ntoOneBack(n);
    }

}

File: python/recursions_backtrackings/day1/basicRec.py
# (1)print Name n times...
# def n_times(i, n):
#     if i > n:    
#         return

#     print("Aman")
#     n_times(i + 1, n)  


# n = int(input("Enter n = "))
# n_times(1, n)




# (2) Print Linearly from 1 to N...
# def linearly(i, n):
#     if i > n:
#         return

#     print(i)
#     linearly(i + 1, n)


# n = int(input("Enter n = "))
# linearly(1, n)






# (3) Print from N to 1
# def n_to_one(n):
#     if n < 1:
#         return

#     print(n)
#     n_to_one(n - 1)


# n = int(input("Enter n = "))
# n_to_one(n)





# (4)Print Linearly (1 to N) using Backtracking
def linearly_back(n):
    if n < 1:
        return

    linearly_back(n - 1)
    print(n)              


n = int(input("Enter n = "))
linearly_back(n)
